std/conv.d

```diff
@@ -5338,3 +5338,236 @@ unittest
+    bool isH;
+    foreach(c; hexData)
+    {
+        isH = c.isHexDigit;
+        if (!isH && !c.isWhite)
+            return false;
+        i += isH;
+    }
+    return !(i & 1) & (i > 0);   
+}
+
+unittest
+{
+    import std.ascii;
+    // empty/whites
+    static assert( !"".isHexLiteral);
+    static assert( !" \r".isHexLiteral);
+    static assert( !whitespace.isHexLiteral);
+    static assert( !""w.isHexLiteral);
+    static assert( !" \r"w.isHexLiteral);
+    static assert( !""d.isHexLiteral);
+    static assert( !" \r"d.isHexLiteral); 
+    // odd x strings
+    static assert( !("5" ~ whitespace).isHexLiteral);
+    static assert( !"123".isHexLiteral);
+    static assert( !"1A3".isHexLiteral);
+    static assert( !"1 23".isHexLiteral);
+    static assert( !"\r\n\tC".isHexLiteral);
+    static assert( !"123"w.isHexLiteral);
+    static assert( !"1A3"w.isHexLiteral);
+    static assert( !"1 23"w.isHexLiteral);
+    static assert( !"\r\n\tC"w.isHexLiteral);
+    static assert( !"123"d.isHexLiteral);
+    static assert( !"1A3"d.isHexLiteral);
+    static assert( !"1 23"d.isHexLiteral);
+    static assert( !"\r\n\tC"d.isHexLiteral);
+    // even x strings with invalid charset
+    static assert( !"12gG".isHexLiteral);
+    static assert( !"2A  3q".isHexLiteral);
+    static assert( !"12gG"w.isHexLiteral);
+    static assert( !"2A  3q"w.isHexLiteral);
+    static assert( !"12gG"d.isHexLiteral);
+    static assert( !"2A  3q"d.isHexLiteral);
+    // valid x strings
+    static assert( ("5A" ~ whitespace).isHexLiteral);
+    static assert( ("5A 01A C FF de 1b").isHexLiteral);
+    static assert( ("0123456789abcdefABCDEF").isHexLiteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF").isHexLiteral);
+    static assert( ("5A 01A C FF de 1b"w).isHexLiteral);
+    static assert( ("0123456789abcdefABCDEF"w).isHexLiteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"w).isHexLiteral);
+    static assert( ("5A 01A C FF de 1b"d).isHexLiteral);
+    static assert( ("0123456789abcdefABCDEF"d).isHexLiteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"d).isHexLiteral);
+    // library version allows what's pointed by issue 10454
+    static assert( ("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF").isHexLiteral);
+}
+
+/**
+The $(D hexString) function is intended to replace the hexadecimal literal strings
+starting with $(D 'x'), which could be removed to simplify the core language.
+
+The function takes a string made of hexadecimal digits and it returns
+the matching $(D string) by converting each pair of digit to a character.
+The input string can also include some ASCII white characters, which can be used
+to keep the literal string readable in the source code.
+
+Params:
+hexData = the $(D string) to be converted.
+C = character type of the output string. 
+
+Returns:
+Either a $(D char[]), a $(D wchar[]) or a $(D dchar[]), according to $(D_PARAM C).
+ */
+@property @safe nothrow pure
+auto hexString(string hexData, C = char)()
```

Comment:
Walter seems insistent that there be a drop-in replacement for `x"..."`. I don't think requiring a cast would meet his expectations.